lib/src/arithmetic/compound_adder.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// compound_adder.dart
+// Implementation of Compund Integer Adder Module 
+// (Output Sum and Sum1 which is Sum + 1). 
+//
+// 2024 September
+// Author: Anton Sorokin <anton.a.sorokin@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// An abstract class for all compound adder module implementations.
+abstract class CompoundAdder extends Adder {
+
+  /// The addition results (+1) in 2s complement form as [sum1]
```

Comment:
This could be clarified, isn't this always equal to sum + 1?  Referring to
```
[sum] + 1
```
might help with this doc comment

Reply:
modified comments